poll.go

```diff
@@ -60,7 +60,7 @@ func (h *Headscale) PollNetMapHandler(ctx *gin.Context) {
 		return
 	}
 
-	machine, err := h.GetMachineByMachineKey(machineKey)
+	machine, err := h.GetValidMachineByMachineKey(machineKey)
```

Comment:
I think that instead of filtering out the machine, we should check `.isExpired` here and return an error that describes the issue.

If we filter it out, we will return a "Record not found" which is not correct, the record is found, but it has expired. 

If think an if after all the error checking ,and then returning if it has expired makes the most sense.

Reply:
The issue is that logout and re-auth results in 2 records for the same machine. One expired and the other not. GetMachineByKey() picks the first machine record from the query result which happens to be the expired one. Another approach would be to refresh the expiration of the expired record when re-authenticating rather than creating a new one. This was how it was handled before the code was refactored.